### DISCUSSION TITLE
Is it possible to filter data? In the demo it ends up bringing everything.

### DISCUSSION THREAD

**Discussion by johnbh3** [SchneiderInfosystems/FB4D#168] [General]
Is it possible to filter data? In the demo it ends up bringing everything.?
![image](https://github.com/SchneiderInfosystems/FB4D/assets/39352208/7cfb7564-5baa-4d7f-a023-e4d1243f92dd)
In this case, only bring what is 1

**Comment by SchneiderInfosystems** [comment]
Good point! This should be possible by using the "equalTo" query parameter (`cGetQueryParamEqualTo`). Unfortunately, I have not already implemented a helper function in TQueryParamsHelper to build a such query. I would expect this could be done similar to the method `TQueryParamsHelper.AddOrderBy`.

https://firebase.google.com/docs/database/rest/retrieve-data

Can you try the following approach? 

```
function TQueryParamsHelper.AddOrderByAndEqualTo(const FieldName,
  FilterValue: string): TQueryParams;
begin
  if not FieldName.IsEmpty then
    Add(cGetQueryParamOrderBy,
      ['"' + StringReplace(FieldName, '"', '""', [rfReplaceAll]) + '"']);
  if not FilterValue.IsEmpty then
    Add(cGetQueryParamEqualTo,
      ['"' + StringReplace(FilterValue, '"', '""', [rfReplaceAll]) + '"']);
  result := self;
end;

var
  Val: TJSONValue;
  Query: TQueryParams;
 begin
   Query := TQueryParams.CreateQueryParams.AddOrderByAndEqualTo('codigoWeb', '1');
   Val := IRealTimeDB.GetSynchronous('cliente', Query);
 end;
```
If it works I would add it and document the new functionality.

**Reply by SchneiderInfosystems** [reply]
I have added now the code for testing. Please inform about your test.

**Comment by SchneiderInfosystems** [comment]
This feature request has now been implemented. Check for usage the wiki
https://github.com/SchneiderInfosystems/FB4D/wiki/FB4D-Reference-IRealTimeDB#filter-by-a-specific-field

Faster feedback would help me with the implementation, but of course I also understand that not all projects can be processed with the same priority. That's why I would like to take this opportunity again. If you are working on a commercial project, sponsoring helps to keep this library alive. If you are working on a project with no commercial goal, awarding a star will at least help to increase the visibility of this project.
